prediction_market_agent/agents/utils.py

```diff
@@ -108,3 +110,26 @@ def memories_to_learnings(memories: list[DatedChatMessage], model: str) -> str:
         prompt_template=prompt,
         model=model,
     )
+
+
+def get_event_date_from_question(question: str) -> datetime | None:
+    llm = ChatOpenAI(
+        model="gpt-4-turbo",
+        temperature=0.0,
+        api_key=APIKeys().openai_api_key.get_secret_value(),
+    )
+    event_date_str = str(
+        llm.invoke(
+            f"Extract the event date in the format `%m-%d-%Y` from the following question, don't write anything else, only the event date in the given format: `{question}`"
```

Comment:
We need to extract this from the question itself, instead of looking at market's closing date, because on Omen, recommendation is to close market 7 days after the outcome is known.

Reply:
Suggestion - use Pydantic output parser for safer parsing (https://python.langchain.com/v0.1/docs/modules/model_io/output_parsers/types/pydantic/).